packages/lit-dev-content/site/docs/tools/server-side-rendering.md

```diff
@@ -0,0 +1,220 @@
+---
+title: Server-side Rendering
+eleventyNavigation:
+  key: Server-side rendering
+  parent: Tools
+  order: 9
+---
+
+Server-side rendering (SSR) Lit templates and elements is possible with the [`@lit-labs/ssr`](/docs/libraries/labs/) package. As with all `@lit-labs` packages, it is considered experimental and not ready for production use but we encourage you to try it out and file any issues for bugs or feature requests.
```

Comment:
I'd separate the labs notice. Do we have a standard alert formatting and messaging for it? @arthurevans

Reply:
We don't, really, because we don't have any other separate pages for specific labs projects, yet. 

Some individual labs packages have their own notices, but they're very package-specific. For example, Lit Virtualizer identifies it as being in "late prerelease:" https://github.com/lit/lit/blob/main/packages/labs/virtualizer/README.md#lit-labsvirtualizer

My understanding was that @kevinpschaaf was working on defining standard stages for labs projects, which might inform the language in the labs alert. 

I'd suggest something like this:

```
{% alert "info" %}

Pre-release software.

Lit labs packages are published to get feedback from the wider community. This code shouldn't be used in production, but we encourage you to try it out and [file issues](link) for any bugs you find. For general feedback, please use the GitHub [discussion](link?).

For more information about the Lit labs process, see [A public facing version of Kevin's doc that doesn't exist yet](future link).

{% endalert %}
```

It might also be nice to add a sentence describing the current stability of the package. But we'd probably want to standardize that a bit and Kevin's lifecycle doesn't have any gradations like that, just "Development" (where we wouldn't have any docs on the site at all), "Evaluation," and then full release. So any package that needs a labs alert would be in the Evaluation stage.


/cc @graynorton and @kevinpschaaf on language for the Labs packages.

Reply:
As I split the SSR docs to multiple pages, this `aside` would only be in the SSR Overview page. I think we talked about potentially utilizing a banner like the one for when viewing v1 docs when the page contents relate to Lit Labs, though I feel like that's maybe a little too much attention drawing.

We could consider adding a little test flask icon to the nav menu for experimental things like MDN does.
![image](https://user-images.githubusercontent.com/40413829/180100553-f3afd90d-f95a-4e36-aa78-f627c9e5e846.png)

Though then the question for our case is should it be added to the "Server-side Rendering" section title, or the individual page titles, or all.

Reply:
I tried adding test tube emojis in front of each section but don't really like how they look. Putting it on the "Server-side rendering" heading made it too long to fit on the side navigation. :/ Open to any suggestions.

Reply:
Maybe we could set `word-break: break-word` on those headings so that it can look like this?

![image](https://user-images.githubusercontent.com/48894/182969289-9f7dbf33-0f4f-4612-979f-5cc77d9e0185.png)

Reply:
On the alert box at the top of each article: when you expand it, it looks like it's part of the main article body, which doesn't look quite right to me. I would maybe just not have the expansion, and shorten the text?

How about this?

```
<icon> SSR support is in <b>Lit Labs</b>, so it's in active early development and may
have bugs/missing features. If you find a bug, file an [issue]. For questions/comments,
start a [discussion].
```

I also softened the language a bit, because do we really want to assert that it shouldn't be used for production? I kinda think we want people to use it, and it's not much use if you can't use it in production. To me that wording suggests it has some really bad known bugs or performance issues, which isn't the case.

For styling, how about we use the existing `alert` class that we have, e.g. like the "info" one at the top of this page: https://pr880-2bb699c---lit-dev-5ftespv5na-uc.a.run.app/docs/components/decorators/

We also have a beaker svg at https://lit.dev/images/alerts/labs.svg which I think is a more clear "labs" icon than the test tube.

Reply:
For the icon, we're going to want more control than an emoji - which can render differently on different screens. I would probably use the Material "science" icon here: https://fonts.google.com/icons?icon.query=science

As for placement... maybe a monotone icon will look ok repeated? Or we live with the wrapping of the section title?

Reply:
> Putting it on the "Server-side rendering" heading made it too long to fit on the side navigation. :/ Open to any suggestions.

How about just "🧪 SERVER RENDERING"?

Reply:
I like that. I think I like the icon to be after the text though. Thoughts?

<p float="left">
<img width="249" alt="image" src="https://user-images.githubusercontent.com/40413829/184037969-325a25da-c1fa-4fcf-9e98-dfcd844b1816.png">
<img width="251" alt="image" src="https://user-images.githubusercontent.com/40413829/184038079-e74cebe2-a10f-4531-820a-74b75c2a4870.png">
</p>

Reply:
This looks good to me!

Reply:
LGTM!